### PR TITLE
Initial IA harvester

### PR DESCRIPTION
**Pull request by moltude**
Internet Archive harvester. Pulls from the Search API

### CODE REVIEW

File: src/main/scala/dpla/ingestion3/harvesters/api/IaHarvester.scala [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+            error.errorSource.url.getOrElse("Undefined url"),
+            error.errorSource.queryParams,
+            error.message
+          ))
+          continueHarvest = false
+        // Handle a successful response
+        case src: ApiSource with ApiResponse =>
+          src.text match {
+            case Some(docs) =>
+              val json = parse(docs)
```
Comment (AudreyAltman) [inline src/main/scala/dpla/ingestion3/harvesters/api/IaHarvester.scala]: Do you want to handle the possibility here that `parse(docs)` will throw an error if IA's JSON is malformed?  Or will it just bubble up in some useful way?  May not be worth worrying about if it's a low risk.  For that matter, is there a chance of an exception being thrown if the `identifier` and `cursor` aren't present where you expect them in the next few lines?

File: src/main/scala/dpla/ingestion3/profiles/ProviderProfiles.scala [hunk 2]
```diff
@@ -36,6 +36,15 @@ class EsdnProfile extends XmlProfile {
   override def getMapping = new EsdnMapping
 }
 
+/**
+  * Internet Archive
+  */
+class IaProfile extends JsonProfile {
+  type Mapping = CdlMapping // FIXME
```
Comment (AudreyAltman) [inline src/main/scala/dpla/ingestion3/profiles/ProviderProfiles.scala]: Do you want to keep `CdlMapping` on lines 43 and 46 for now?

File: src/main/scala/dpla/ingestion3/harvesters/api/IaHarvester.scala [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+
+              // Loop control
+              cursor = (json \\ "cursor").extractOrElse[String]("")
+
+              if (cursor.isEmpty)
+                continueHarvest = false
+            // Handle unknown case
+            case _ =>
+              harvestLogger.error(s"Response body is empty.\n" +
+                s"URL: ${src.url.getOrElse("!!! URL not set !!!")}\n" +
```
Comment (AudreyAltman) [inline src/main/scala/dpla/ingestion3/harvesters/api/IaHarvester.scala]: I like the exclamation marks :)

File: src/main/scala/dpla/ingestion3/profiles/ProviderProfiles.scala (hunk 2 above)
[the remark this answers is not in the document]
Comment (moltude) [reply src/main/scala/dpla/ingestion3/profiles/ProviderProfiles.scala]: Yes, this is just a place holder until the Mapping is written. `type Mapping = ???` isn't valid and using a stub like this is the only work around that I've found so far.

Comment (moltude) [reply src/main/scala/dpla/ingestion3/profiles/ProviderProfiles.scala]: Yes, see above

File: src/main/scala/dpla/ingestion3/harvesters/api/IaHarvester.scala (hunk 1 above)
Comment (moltude) [reply src/main/scala/dpla/ingestion3/harvesters/api/IaHarvester.scala]: I added some additional error handling in the latest commit

**Comment by moltude** [comment]
@AudreyAltman bumping for changes

File: src/main/scala/dpla/ingestion3/profiles/ProviderProfiles.scala (hunk 2 above)
[the remark this answers is not in the document]
Comment (mdellabitta) [reply src/main/scala/dpla/ingestion3/profiles/ProviderProfiles.scala]: Maybe we just need an EvilMapping that throws for every method called on it as a placeholder.

**Pull request merged**